chia/plotting/manager.py

```diff
@@ -224,7 +288,7 @@ def process_file(file_path: Path) -> Dict:
                     else:
                         assert isinstance(pool_public_key_or_puzzle_hash, bytes32)
                         pool_public_key = None
-                        pool_contract_puzzle_hash = pool_public_key_or_puzzle_hash
+                        pool_contract_puzzle_hash = bytes(pool_public_key_or_puzzle_hash)
```

Comment:
This messes with the type which should actually be stored in `PlotInfo` for `pool_contract_puzzle_hash` which should be `bytes32` not `bytes` but i couldn't get `bytes32` pickable..  i tried adding the `__setstate__` and `__getstate__` to its creation in `byte_types.py` which then have definitely been called when pickling but pickling still didn't work. Im not sure if they were correct because i obviously miss some python background here. Error was with and without those methods:
```
_pickle.PicklingError: Can't pickle <class 'chia.util.byte_types.bytes32'>: attribute lookup bytes32 on chia.util.byte_types failed
```

Anyone has an idea here?

Reply:
Two options I can see:

- fix `pickle` so it knows how to serialize `bytes32`. The source for the `pickle` module is [here](https://github.com/python/cpython/blob/main/Lib/pickle.py) and [here](https://github.com/python/cpython/blob/main/Modules/_pickle.c). This is a very old, very legacy module, so it's a bit tricky to follow.
- use a different serialization besides `pickle`, like `json`. We already use `Streamable` to recursively turn into JSON in a lot of places, so this might be the path of least resistance.

I've always been a bit reticent about using `pickle` (except for quick hacks) for reasons I can't fully quantify, so I don't have a lot of experience with this module.

Reply:
It might be enough to implement [this](https://docs.python.org/3/library/pickle.html#object.__getnewargs__) on `bytes32` to fix the pickle problem. Let me try.

Reply:
Okay, that didn't work. I think the error message is a result of `pickle` trying to do `from chia.util.byte_types.bytes32 import bytes32` which doesn't work because that's where the `byte_types` template is defined, but not the class itself (it's in `chia.types.blockchain_format.sized_bytes`).

Reply:
Thanks again @richardkiss helping to figure out the hacky `bytes32` pickle support added by f9f603e.